### PR TITLE
Part of mandatory tests 2

### PR DESCRIPTION
**Pull request by domachine**
HEADS UP! This PR bases on the first part (#186) therefore the diff is a little bit messy but will be easier to read after merging the first part. Furthermore I started to extract the tests into their own modules.

### CODE REVIEW

**Comment by domachine** [comment]
@tschmidtb51 Thank you very much for your review. I addressed the issues and added another test to make sure that "6.1.7" is correct.

**Review by sthagen — APPROVED** [review]
Nice.

File: app/lib/shared/Core/entities/DocumentEntity/mandatoryTest_6_1_23.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  if (preconditionFor_6_1_23_Matches(doc)) {
+    /** @type {Set<string>} */
+    const cveStrings = new Set()
+
+    doc.vulnerabilities.forEach((vulnerability, vulnerabilityIndex) => {
+      if (vulnerability.cve === undefined) return
+      if (cveStrings.has(vulnerability.cve)) {
+        isValid = false
+        errors.push({
+          message: `tracking-number was already used`,
```
[the remark this answers is not in the document]
Comment (tschmidtb51) [reply app/lib/shared/Core/entities/DocumentEntity/mandatoryTest_6_1_23.js]: Resolved in cb19b0730.

File: app/lib/shared/Core/entities/DocumentEntity/mandatoryTest_6_1_24.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
+/**
+ * @param {unknown} doc
+ */
+export default function mandatoryTest_6_1_24(doc) {
+  /** @type {Array<{ message: string; dataPath: string }>} */
+  const errors = []
+  let isValid = true
+
+  // 6.1.24 Multiple Use of Same CVE
```
[the remark this answers is not in the document]
Comment (tschmidtb51) [reply app/lib/shared/Core/entities/DocumentEntity/mandatoryTest_6_1_24.js]: Resolved in cb19b073.

File: app/tests/shared/coreFixture.js [hunk 3]
```diff
@@ -0,0 +1,663 @@
...
+          {
+            scores: [
+              {
+                products: ['CSAFPID-9080700'],
+                cvss_v2: {
+                  version: '2.0',
+                  vectorString: 'CVSS:2.0/AV:N/AC:L/PR:N/UI:N/S:C/C:H/I:H/A:H',
+                  baseScore: 10,
+                  baseSeverity: 'CRITICAL',
+                },
```
[the remark this answers is not in the document]
Comment (tschmidtb51) [reply app/tests/shared/coreFixture.js]: Fixed in cb19b0730a8f.

File: app/tests/shared/coreFixture.js [hunk 4]
```diff
@@ -0,0 +1,663 @@
...
+              },
+              {
+                products: ['CSAFPID-9080700'],
+                cvss_v2: {
+                  version: '2.0',
+                  vectorString: 'CVSS:2.0/AV:L/AC:L/PR:H/UI:R/S:U/C:H/I:H/A:H',
+                  baseScore: 6.5,
+                  baseSeverity: 'MEDIUM',
+                },
```
[the remark this answers is not in the document]
Comment (tschmidtb51) [reply app/tests/shared/coreFixture.js]: Resolved in cb19b0730.

File: app/lib/shared/Core/entities/DocumentEntity/mandatoryTest_6_1_7.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+              Array.isArray(score.products) &&
+              score.products.every((product) => typeof product === 'string') &&
+              ((score.cvss_v2 &&
+                (typeof score.cvss_v2 === 'string' ||
+                  score.cvss_v2 === undefined)) ||
+                score.cvss_v2 === undefined) &&
+              ((score.cvss_v3 &&
+                (typeof score.cvss_v3 === 'string' ||
+                  score.cvss_v3 === undefined)) ||
+                score.cvss_v2 === undefined)
```
[the remark this answers is not in the document]
Comment (tschmidtb51) [reply app/lib/shared/Core/entities/DocumentEntity/mandatoryTest_6_1_7.js]: Resolved in cb19b0730.

File: app/lib/shared/Core/entities/DocumentEntity/mandatoryTest_6_1_7.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  const errors = []
+  let isValid = true
+
+  // 6.1.7 Multiple Scores with same Version per Product
+  if (preconditionFor_6_1_7_Matches(doc)) {
+    /** @type {Map<string, Set<string>>} */
+    const cvssVersionsByProductName = new Map()
+
+    doc.vulnerabilities.forEach((vulnerability, vulnerabilityIndex) => {
+      vulnerability.scores?.forEach((score, scoreIndex) => {
```
[the remark this answers is not in the document]
Comment (tschmidtb51) [reply app/lib/shared/Core/entities/DocumentEntity/mandatoryTest_6_1_7.js]: Resolved in cb19b0730.

**Review by tschmidtb51 — APPROVED** [review]
Perfect. Ready to merge.

**Comment by bentolor** [comment]
Thanks. Merged with e3d6cc0

**Pull request merged**